core/collateral/engine.go

```diff
@@ -2869,3 +2876,244 @@ func (e *Engine) GetRewardAccountsByType(rewardAcccountType types.AccountType) [
 	}
 	return accounts
 }
+
+// TransferToHoldingAccount locks funds from general account into holding account account of the party.
+func (e *Engine) TransferToHoldingAccount(ctx context.Context, transfer *types.Transfer) (*types.LedgerMovement, error) {
+	generalAccountID := e.accountID(transfer.Market, transfer.Owner, transfer.Amount.Asset, types.AccountTypeGeneral)
+	generalAccount, err := e.GetAccountByID(generalAccountID)
+	if err != nil {
+		return nil, err
+	}
+
+	holdingAccountID := e.accountID(noMarket, transfer.Owner, transfer.Amount.Asset, types.AccountTypeHolding)
+	holdingAccount, err := e.GetAccountByID(holdingAccountID)
+	if err != nil {
+		// if the holding account doesn't exist yet we create it here
+		holdingAccountID, err := e.CreatePartyHoldingAccount(ctx, transfer.Owner, transfer.Amount.Asset)
+		if err != nil {
+			return nil, err
+		}
+		holdingAccount, _ = e.GetAccountByID(holdingAccountID)
+	}
+
+	req := &types.TransferRequest{
+		Amount:      transfer.Amount.Amount.Clone(),
+		MinAmount:   transfer.Amount.Amount.Clone(),
+		Asset:       transfer.Amount.Asset,
+		Type:        types.TransferTypeHoldingAccount,
+		FromAccount: []*types.Account{generalAccount},
+		ToAccount:   []*types.Account{holdingAccount},
+	}
+
+	res, err := e.getLedgerEntries(ctx, req)
+	if err != nil {
+		e.log.Error("Failed to transfer funds", logging.Error(err))
+		return nil, err
+	}
+	for _, bal := range res.Balances {
+		if err := e.IncrementBalance(ctx, bal.Account.ID, bal.Balance); err != nil {
```

Comment:
Not sure if we should call `IncrementBalance`. The `getLedgerEntries` updates the account balance with stuff like: `to.Account.Balance.AddSum(remainder)`, so wouldn't `UpdateBalance` make more sense?

In most cases we do use `IncrementBalance`, but the amount by which we increment is taken from `res.Entries[].Amount`